cpp/tests/utilities_tests/column_wrapper_tests.cu

```diff
@@ -185,3 +198,689 @@ TYPED_TEST(FixedWidthColumnWrapperTest, ReleaseWrapperAllNull)
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2, 4});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto child = lcv.child();
+    lists_column_view childv(child);
+    EXPECT_EQ(childv.size(), 4);
+    EXPECT_EQ(childv.null_count(), 1);
+
+    auto child_offsets = childv.offsets();
+    EXPECT_EQ(child_offsets.size(), 5);
+    test::fixed_width_column_wrapper<size_type> e_child_offsets({0, 2, 2, 4, 6});
+    test::expect_columns_equal(e_child_offsets, child_offsets);
+
+    auto child_child = childv.child();
+    lists_column_view child_childv(child_child);
+    EXPECT_EQ(child_childv.size(), 6);
+
+    auto child_child_offsets = child_childv.offsets();
+    EXPECT_EQ(child_child_offsets.size(), 7);
+    test::fixed_width_column_wrapper<size_type> e_child_child_offsets({0, 2, 4, 6, 8, 11, 12});
+    test::expect_columns_equal(e_child_child_offsets, child_child_offsets);
+
+    auto child_child_data = child_childv.child();
+    EXPECT_EQ(child_child_data.size(), 12);
+    test::fixed_width_column_wrapper<int> e_child_child_data(
+      {1, 2, 3, 4, -1, -2, -3, -4, -5, -6, -7, 0});
+    test::expect_columns_equal(child_child_data, child_child_data);
+  }
+}
+
+TEST_F(ListColumnWrapperTest, ListTypesMismatch)
+{
+  using namespace cudf;
+
+  using L  = std::initializer_list<int>;
+  using L2 = std::initializer_list<float>;
+
+  {
+    auto should_throw = []() { test::lists_column_wrapper list{L{2, 3}, L2{4, 5}}; };
+    EXPECT_THROW(should_throw(), cudf::logic_error);
+  }
+
+  {
+    auto should_throw = []() {
+      test::lists_column_wrapper list{{L{2, 3}, L{4, 5}}, {L2{6, 7}, L2{8, 9}}};
+    };
+    EXPECT_THROW(should_throw(), cudf::logic_error);
+  }
+}
+
+TYPED_TEST(ListColumnWrapperTestTyped, ListOfType)
+{
+  using namespace cudf;
+
+  using T = TypeParam;
+  using L = std::initializer_list<T>;
+
+  // List<T>, 1 row
+  {
+    test::lists_column_wrapper list{L{2, 3}};
+    lists_column_view lcv(list);
+    EXPECT_EQ(lcv.size(), 1);
+
+    auto offsets = lcv.offsets();
+    EXPECT_EQ(offsets.size(), 2);
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto data = lcv.child();
+    EXPECT_EQ(data.size(), 2);
+    test::fixed_width_column_wrapper<T> e_data({2, 3});
+    test::expect_columns_equal(e_data, data);
+  }
+
+  // List<T>, 3 rows
+  {
+    test::lists_column_wrapper list{L{2, 3}, L{4, 5}, L{6, 7, 8}};
+    lists_column_view lcv(list);
+    EXPECT_EQ(lcv.size(), 3);
+
+    auto offsets = lcv.offsets();
+    EXPECT_EQ(offsets.size(), 4);
+    test::fixed_width_column_wrapper<size_type> e_offsets({0, 2, 4, 7});
+    test::expect_columns_equal(e_offsets, offsets);
+
+    auto data = lcv.child();
+    EXPECT_EQ(data.size(), 7);
+    test::fixed_width_column_wrapper<T> e_data({2, 3, 4, 5, 6, 7, 8});
+    test::expect_columns_equal(e_data, data);
+  }
+}
+
+TYPED_TEST(ListColumnWrapperTestTyped, ListOfListOfTypes)
```

Comment:
If you have typed tests for lists of types (And nested lists of types), then why do you need the same tests for just ints above?

Reply:
I wanted to distinguish between:
- Inferring the type based on the specified literals.
- Explicitly casting to specific type.

They are somewhat redundant, but it's definitely cleaner when you can declare these things without any casting (they're kind of eye-bending to read already) so I figured I'd just be explicit in the testing.

Reply:
Can you add a comment to this effect?